src/Aquifer.Common/Services/ResourceContentSearchService.cs

```diff
@@ -0,0 +1,672 @@
+﻿using Aquifer.Data;
+using Aquifer.Data.Entities;
+using Dapper;
+using Microsoft.EntityFrameworkCore;
+using static Aquifer.Common.Services.ResourceContentSearchResult;
+
+namespace Aquifer.Common.Services;
+
+public interface IResourceContentSearchService
+{
+    Task<(int Total, IReadOnlyList<ResourceContentSearchResult> ResourceContentSummaries)> SearchAsync(
+        ResourceContentSearchIncludeFlags includeFlags,
+        ResourceContentSearchFilter filter,
+        ResourceContentSearchSortOrder sortOrder,
+        int offset,
+        int? limit,
+        CancellationToken ct);
+}
+
+public sealed class ResourceContentSearchFilter
+{
+    public int? ParentResourceId { get; set; }
+    public int? ResourceId { get; set; }
+    public string? ResourceEnglishLabelQuery { get; set; }
+    public int? LanguageId { get; set; }
+    public IReadOnlyList<ResourceContentMediaType>? ExcludeContentMediaTypes { get; set; }
+    public IReadOnlyList<ResourceContentMediaType>? IncludeContentMediaTypes { get; set; }
+    public IReadOnlyList<ResourceContentStatus>? ExcludeContentStatuses { get; set; }
+    public IReadOnlyList<ResourceContentStatus>? IncludeContentStatuses { get; set; }
+    public int? AssignedUserId { get; set; }
+    public int? AssignedUserCompanyId { get; set; }
+    public bool? IsPublished { get; set; }
+    public bool? IsDraft { get; set; }
+    public int? StartVerseId { get; set; }
+    public int? EndVerseId { get; set; }
+    public bool? HasAudio { get; set; }
+    public bool? HasUnresolvedCommentThreads { get; set; }
+    public bool? IsInProject { get; set; }
+}
+
+/// <summary>
+/// These things are expensive so you have to specifically request them.
+/// </summary>
+[Flags]
+public enum ResourceContentSearchIncludeFlags
+{
+    None = 0,
+
+    Project = 1,
+    ResourceContentVersions = 2,
+    HasAudioForLanguage = 4,
+    HasUnresolvedCommentThreads = 8,
+
+    All = Project | ResourceContentVersions | HasAudioForLanguage | HasUnresolvedCommentThreads,
+}
+
+public enum ResourceContentSearchSortOrder
+{
+    ResourceContentId = 0,
+
+    /// <summary>
+    /// Sort: ParentResource.DisplayName, Resource.SortOrder, Resource.EnglishLabel, ResourceContent.LanguageId
```

Comment:
I added `LanguageId` here and below for consistent ordering results when multiple languages are present with the same `EnglishLabel`.